cfx_types/src/lib.rs

```diff
@@ -26,3 +27,39 @@ pub fn hexstr_to_h256(hex_str: &str) -> H256 {
 
     H256::from(bytes)
 }
+
+pub trait AddressUtil {
+    fn type_bits(&self) -> u8;
+
+    fn is_valid<T>(&self, builtin_map: &BTreeMap<Address, T>) -> bool;
+
+    fn is_contract(&self) -> bool;
+
+    fn is_payment(&self) -> bool;
+
+    fn is_builtin<T>(&self, builtin_map: &BTreeMap<Address, T>) -> bool;
+
+    fn converted_to_contract(&mut self);
+}
+
+impl AddressUtil for Address {
+    fn type_bits(&self) -> u8 { self.as_fixed_bytes()[0] & 0xf0 }
+
+    fn is_valid<T>(&self, builtin_map: &BTreeMap<Address, T>) -> bool {
+        self.is_contract() || self.is_payment() || self.is_builtin(builtin_map)
+    }
+
+    /// Should keep this consistent with `converted_to_contract()`.
```

Comment:
It's simple to design a structure to accomplish the purpose in your comment:

define consts for 0x80, 0x10, 0x10 etc.
define a method set_type_bits(address: &mut H160, type_bits: u8) {  let first_bytes = &mut address.as_bytes_mut()[0]; *first_bytes &= 0x0f; *first_bytes |= type_bits }
let set_contract_type_bits call set_type_bits